src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -529,6 +529,13 @@ def create_realm_zonegroup_zone(self, spec: RGWSpec, rgw_id: str):
             except Exception as e:
                 raise OrchestratorError('failed to parse zone info')
 
+        cmd = ['radosgw-admin',
+               '--key=%s'%keyring,
+               '--user', 'rgw.%s'%rgw_id,
+               'period', 'update',
+               '--rgw-realm=%s'%spec.rgw_realm,
+               '--commit']
+        result = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
would be great to call this, only if we need to. wdyt?

Reply:
agreed. ill move some code around so we only do this if we are creating the realm/zone